sdks/aperture-go/sdk/middlewares/grpc.go

```diff
@@ -0,0 +1,144 @@
+package middlewares
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	flowcontrolhttp "github.com/fluxninja/aperture-go/v2/gen/proto/flowcontrol/checkhttp/v1"
+	aperture "github.com/fluxninja/aperture-go/v2/sdk"
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+// socketAddressFromNetAddr takes a net.Addr and returns a flowcontrolhttp.SocketAddress.
+func socketAddressFromNetAddr(logger logr.Logger, addr net.Addr) *flowcontrolhttp.SocketAddress {
+	host, port := splitAddress(logger, addr.String())
+	protocol := flowcontrolhttp.SocketAddress_TCP
+	if addr.Network() == "udp" {
+		protocol = flowcontrolhttp.SocketAddress_UDP
+	}
+	return &flowcontrolhttp.SocketAddress{
+		Address:  host,
+		Protocol: protocol,
+		Port:     port,
+	}
+}
+
+// GRPCUnaryInterceptor takes a control point name and creates a UnaryInterceptor which can be used with gRPC server.
+func GRPCUnaryInterceptor(c aperture.Client, controlPoint string) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		labels := aperture.LabelsFromCtx(ctx)
+
+		md, ok := metadata.FromIncomingContext(ctx)
+		if ok {
+			for key, value := range md {
+				labels[key] = strings.Join(value, ",")
+			}
+		}
+
+		var sourceSocket *flowcontrolhttp.SocketAddress
+		if sourceAddr, ok := peer.FromContext(ctx); ok {
+			sourceSocket = socketAddressFromNetAddr(c.GetLogger(), sourceAddr.Addr)
+		}
+		// TODO: Can we retrieve the address somehow?
+		var destinationSocket *flowcontrolhttp.SocketAddress
```

Comment:
We need this address in CheckHTTP otherwise the SDK won't work if someone has added a service selector in policy.
CheckHTTP uses destination address to get the target service

Reply:
Ref - https://github.com/fluxninja/aperture/blob/main/pkg/policies/flowcontrol/service/checkhttp/checkhttp.go#L174

Reply:
Updated.